jni_gen/lib/src/config/config_provider.dart

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:yaml/yaml.dart';
+
+class ConfigError extends Error {
+  ConfigError(this.message);
+  final String message;
+  @override
+  String toString() => message;
+}
+
+class ConfigProvider {
+  ConfigProvider.of(this.cli, this.yaml);
+  Map<String, String> cli;
+  Map<dynamic, dynamic> yaml;
+
+  static ConfigProvider parseArgs(List<String> args,
```

Comment:
Let `parseArgs` return a `Config` object.

Similar to https://github.com/dart-lang/ffigen/blob/master/lib/src/config_provider/config.dart#L21.

This enables us to:

1. catch config errors before doing any code generation
2. write code generation tests by passing in a `Config` we just set the fields of in Dart code instead of through command line args or yaml.